File: cpp/opengl/src/vsdk/toolkit/render/opengl/visualDebugTools/PatchHitCandidate.cpp
#include "vsdk/toolkit/render/opengl/visualDebugTools/PatchHitCandidate.h"

PatchHitCandidate::PatchHitCandidate():
    patchIndex(-1),
    distance(0.0F),
    frontFacing(false)
{
}

PatchHitCandidate::PatchHitCandidate(int patchIndex, float distance, bool frontFacing):
    patchIndex(patchIndex),
    distance(distance),
    frontFacing(frontFacing)
{
}

File: java/base/src/main/java/vsdk/toolkit/render/jogl/visualDebugTools/PatchHitCandidate.java
package vsdk.toolkit.render.jogl.visualDebugTools;

public class PatchHitCandidate {
    public int patchIndex;
    public float distance;
    public boolean frontFacing;

    public PatchHitCandidate() {
        this(-1, 0.0f, false);
    }

    public PatchHitCandidate(int patchIndex, float distance, boolean frontFacing) {
        this.patchIndex = patchIndex;
        this.distance = distance;
        this.frontFacing = frontFacing;
    }
}
